src/components/profile/VerifyEditCode.js

```diff
@@ -0,0 +1,184 @@
+// @flow
+import React, { useEffect, useState } from 'react'
+import _get from 'lodash/get'
+import logger from '../../lib/logger/pino-logger'
+import API from '../../lib/API/api'
+import { getDesignRelativeHeight } from '../../lib/utils/sizes'
+import { withStyles } from '../../lib/styles'
+import SMSFormComponent from '../signup/SMSFormComponent'
+import { useDialog } from '../../lib/undux/utils/dialog'
+import userStorage from '../../lib/gundb/UserStorage'
+
+const log = logger.child({ from: 'SmsForm' })
+
+export type SMSRecord = {
+  smsValidated: boolean,
+  sentSMS?: boolean,
+}
+
+const NumInputs: number = 6
+
+const VerifyEditCode = props => {
```

Comment:
why dont we use SMSForm and EmailConfirmation components directly? why do we need new components?

Reply:
@sirpy I made SMSForm markup be possible to use in other places; separated it to other component.  So the VerifyEditCode uses the same SMSFormComponent as the SMSForm does.

The SMSForm and EmailConfirmation definitely have the same markup, there are only text differences.

Reply:
yes but why? why not use the exact same components?

Reply:
@sirpy because I am using different functionality (sending other API requests, showing popup on success, etc.) which is not the same functionality as the signup SMSForm does.

Reply:
@yaroslav-fedyshyn-nordwhale 
still I dont see a reason.
1. still think we should consolidate to the same API (see my comment in server pull request)
2. liav asked we dont show a popup
3. in any case we have the "done" callback we can use to show popup